AmplifyPlugins/API/Tests/APIHostApp/GraphQLAPIStressTests/GraphQLAPIStressTests.swift

```diff
@@ -0,0 +1,355 @@
+                                    XCTFail("\(error)")
+                                }
+                            }
+                        }
+                        await completedInvoked.fulfill()
+                    }
+                    return subscription
+                }
+            }
+            
+            var sequences = [AmplifyAsyncThrowingSequence<GraphQLSubscriptionEvent<Post>>]()
+            for await sequence in group {
+                sequences.append(sequence)
+            }
+            return sequences
+            
+        }
+    
+        await waitForExpectations([connectedInvoked], timeout: TestCommonConstants.networkTimeout)
+        XCTAssertEqual(sequences.count, concurrencyLimit)
+        
+        guard try await createPost(id: uuid, title: title) != nil else {
+            XCTFail("Failed to create post")
+            return
+        }
+
+        await waitForExpectations([progressInvoked], timeout: TestCommonConstants.networkTimeout)
+        sequences.forEach { sequence in
+            sequence.cancel()
+        }
+        await waitForExpectations([disconnectedInvoked, completedInvoked], timeout: TestCommonConstants.networkTimeout)
+    }
+    
+    /// - Given: APIPlugin configured with valid configuration and schema
+    /// - When: I create 50 posts simultaneously
+    /// - Then: Operation should succeed
+    func testMultipleCreateMutations() async throws {
+        let postCreateExpectation = asyncExpectation(description: "Post was created successfully",
+                                                     expectedFulfillmentCount: concurrencyLimit)
+        for index in 0..<concurrencyLimit {
+            Task {
+                let id = UUID().uuidString
+                let title = "title" + String(index)
+                let post = try await createPost(id: id, title: title)
+                XCTAssertNotNil(post)
+                XCTAssertEqual(id, post?.id)
+                XCTAssertEqual(title, post?.title)
+                await postCreateExpectation.fulfill()
+            }
+        }
+        
+        await waitForExpectations([postCreateExpectation], timeout: TestCommonConstants.networkTimeout)
+    }
+    
+    /// - Given: APIPlugin configured with valid configuration and schema and 50 posts saved
+    /// - When: I update 50 post simultaneously
+    /// - Then: Operation should succeed
+    func testMultipleUpdateMutations() async throws {
+        let postCreateExpectation = asyncExpectation(description: "Post was created successfully",
+                                                     expectedFulfillmentCount: concurrencyLimit)
+        let postUpdateExpectation = asyncExpectation(description: "Post was updated successfully",
+                                                     expectedFulfillmentCount: concurrencyLimit)
+        let postOffice = PostOffice()
+        
+        for index in 0..<concurrencyLimit {
```

Comment:
Coming from reviewing https://github.com/aws-amplify/amplify-swift/pull/2639/files what's the difference between the different ways to simultaneously ways to update the post? In the DS PR we're using DispatchQueue.concurrentPerform, in this PR we're creating Tasks on each for loop iteration. And then there's the above test that uses a TaskGroup. Which one should we be using for stress testing?

Reply:
`TaskGroup` is used for concurrency for e.g. if there are multiple parts of your code which `async` and related, they can be put within a task group. `.concurrentPerform` is used when there are multiple instances of `async` and unrelated code. In this case, we are running multiple instances of same code on different data. Updated it to use `.concurrentPerform`